ui/app/router.js

```diff
@@ -127,7 +127,9 @@ Router.map(function() {
         this.route('show', { path: '/:policy_name' });
         this.route('edit', { path: '/:policy_name/edit' });
       });
-      this.route('replication-dr-promote');
+      this.route('replication-dr-promote', function() {
+        this.route('details');
```

Comment:
can you refresh me again why we have to put the details route inside `replication-dr-promote`? from a UX perspective, i agree that it makes sense to keep `replication-dr-promote` where it is so as not to interrupt users existing workflows. however semantically `replication` feels like an action to me, and so `replication-dr-promote/details` feels like it would show the details related to that action. i wonder -- could we set up a route structure like this?

```
/vault/replication/
  -- /vault/replication-dr-promote
  -- /vault/replication/dr (checks whether `isSecondary` is true, then defaults to the details tab of the appropriate template)
    -- vault/replication/dr/manage
    -- vault/replication/dr/secondaries
...
// etc., and the performance routes could follow the same pattern
```

i know we discussed this earlier, so apologies for not hashing it out totally then ~ i just needed more time to ponder and think of other options. what do you think?

Reply:
This is the only route where we are accessing the Secondary Disaster Recovery view.  Which also means, by the time they get to this route all the permissions have been setup.  So similar to how we did Metrics.  Once you're in the metrics URL you get access to the other sub-routes.  

```
// Metrics route
this.route('metrics', function() {
        this.route('index', { path: '/' });
        this.route('http-requests');
      });
```

So in this case, you have navigated to the Secondary Disaster Recovery main route, and now you have access to see its details.  No other route gets you to the this view.  Does that make sense?

Matthew, if you have thoughts on this, please chime in.  I could be totally off on this.

Reply:
I'm not sure how difficult it'd be, it'd certainly take changes to `cluster-route.js` - I think the tricky part there would be that you'd have to check the model / params on the transition object because the replication routes have a route param in the URL. Replication is top-level though (https://github.com/hashicorp/vault/blob/6d273dc368b86998d953d8abf06e5e94ab19dba0/ui/app/initializers/enable-engines.js#L7) so it might not be too bad.

Reply:
Thank you for the feedback.  I'll play around with this a bit and see how complex it might be and also chat with Noelle.